bundles/org.openhab.binding.qbus/src/main/java/org/openhab/binding/qbus/internal/protocol/QbusCommunication.java

```diff
@@ -0,0 +1,926 @@
+    private Gson gsonIn;
+    @Nullable
+    private String CTD;
+
+    private Boolean CTDConnected = false;
+
+    private final Map<Integer, QbusBistabiel> bistabiel = new HashMap<>();
+    private final Map<Integer, QbusScene> scene = new HashMap<>();
+    private final Map<Integer, QbusDimmer> dimmer = new HashMap<>();
+    private final Map<Integer, QbusRol> rol = new HashMap<>();
+    private final Map<Integer, QbusThermostat> thermostat = new HashMap<>();
+    private final Map<Integer, QbusCO2> co2 = new HashMap<>();
+
+    @Nullable
+    private QbusBridgeHandler bridgeCallBack;
+
+    /**
+     * Constructor for Qbus communication object, manages communication with
+     * Qbus Server.
+     *
+     */
+    public QbusCommunication() {
+        GsonBuilder gsonBuilder = new GsonBuilder();
+        gsonBuilder.registerTypeAdapter(QbusMessageBase.class, new QbusMessageDeserializer());
+        gsonIn = gsonBuilder.create();
+    }
+
+    /**
+     * Start communication with Qbus Server, run through initialization and start thread listening
+     * to all messages coming from Qbus.
+     *
+     * @param addr : IP-address of Qbus Server
+     * @param port : Communication port of Qbus server
+     *
+     */
+
+    public synchronized void startCommunication() {
+        QbusBridgeHandler handler = bridgeCallBack;
+        CTDConnected = false;
+        try {
+
+            for (int i = 1; qEventsRunning && (i <= 5); i++) {
+                Thread.sleep(1000);
+            }
+            if (qEventsRunning) {
+                logger.error("Starting from thread {}, but previous connection still active after 5000ms",
+                        Thread.currentThread().getId());
+                throw new IOException();
+            }
+
+            if (handler == null) {
+                throw new IOException();
+            }
+
+            InetAddress addr = InetAddress.getByName(handler.getAddress());
+            Integer port = handler.getPort();
+
+            if (port != null) {
+                Socket socket = new Socket(addr, port);
+                qSocket = socket;
+                qOut = new PrintWriter(socket.getOutputStream(), true);
+                qIn = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+                logger.info("Connected via local port {} from thread {}", socket.getLocalPort(),
+                        Thread.currentThread().getId());
+            } else {
+                return;
+            }
+
+            setSN();
+            getSN();
+
+            // Connect to Qbus server
+            Connect();
+
+            // If Qbus Client is connected then initialize, else put Bridge offline
+            if (CTDConnected == true) {
+                initialize();
+                (new Thread(qEvents)).start();
+            } else {
+                handler.bridgeOffline("No communication with Qbus client");
+            }
+        } catch (IOException | InterruptedException e) {
```

Comment:
The InterruptedException is thrown when OH is shut-down. That means the current operation should be finished as fast as possible and no further operations should be done. You can achieve this by not catching the exception here but throw the InterruptedException and catch it at the highest level you can. As this is not an error, it shouldn't be logged, but only handled.